crates/ruff/src/rule_selector.rs

```diff
@@ -233,3 +233,76 @@ pub(crate) enum Specificity {
     Code4Chars,
     Code5Chars,
 }
+
+mod clap_completion {
```

Comment:
I put it in a submodule so that we can easily feature gate it in the future (since we'll probably want to turn `clap` into an optional dependency for the `ruff` library).